jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -766,7 +766,7 @@ private void completeStream(HttpStream stream, Throwable failure)
                 }
 
                 // Clean up any multipart tmp files and release any associated resources.
-                Parts parts = (Parts)_request.getAttribute(Parts.class.getName());
+                MultiPartFormData.Parts parts = MultiPartFormData.getParts(_request);
```

Comment:
`MultiPartFormData.getParts()` may block indefinitely.
Are we sure that by reaching this code, we won't block?

Can you add a test case where the client sends an incomplete part, and verify what happens?

Reply:
I don't think we should be calling this method here - we never want it to parse the parts if they have not already been parsed.